File: Selenium/Activities/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/selects")

print("Page title:", driver.title)

dropdown = driver.find_element(By.CSS_SELECTOR, "select.h-10")

select = Select(dropdown)

select.select_by_visible_text("Two")
print("Selected (visible text):", select.first_selected_option.text)

select.select_by_index(3)
print("Selected (index):", select.first_selected_option.text)

select.select_by_value("four")
print("Selected (value):", select.first_selected_option.text)

print("\nAll options:")
for option in select.options:
    print(option.text)

driver.quit()

File: Selenium/Activities/Activity16.java
package examples;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import io.github.bonigarcia.wdm.WebDriverManager;

public class Activity16 {
    public static void main(String[] args) {

        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/selects");

        System.out.println("Page title is: " + driver.getTitle());

        WebElement dropdown = driver.findElement(By.cssSelector("select.h-10"));

        Select select = new Select(dropdown);

        select.selectByVisibleText("Two");
        System.out.println("Selected (visible text): " + 
            select.getFirstSelectedOption().getText());

        select.selectByIndex(3);
        System.out.println("Selected (index): " + 
            select.getFirstSelectedOption().getText());

        select.selectByValue("four");
        System.out.println("Selected (value): " + 
            select.getFirstSelectedOption().getText());

        List<WebElement> options = select.getOptions();
        System.out.println("\nAll options:");
        for (WebElement option : options) {
            System.out.println(option.getText());
        }

        driver.quit();
    }
}